imports/ui/components/edit/edit-collection.jsx

```diff
@@ -34,69 +36,64 @@ let EditCollection = React.createClass({
     this.setState({ list: newList, lastMoved: newList.length - 1, lastDir: -1 }, this.update)
   },
   storeButton (index, dir) {
-    return (index === this.state.lastMoved && dir === this.state.lastDir) ?
-      button => this._focusButton = button :
-      () => {}
+    return (index === this.state.lastMoved && dir === this.state.lastDir)
+      ? (button) => { this._focusButton = button }
+      : () => {}
   },
   saveAndFollowLink (link, e) {
     this.props.save(e, () => {
-      FlowRouter.go(link)
+      browserHistory.push(link)
     })
   },
   refocus (e) {
     switch (e.keyCode) {
       case 37:
       case 39:
-      this.setState({ lastDir: this.state.lastDir * -1 })
-      return
+        this.setState({ lastDir: this.state.lastDir * -1 })
+        return
 
       case 38:
-      this.setState({ lastMoved: (this.state.lastMoved - 1) % this.state.list.length })
-      return
+        this.setState({ lastMoved: (this.state.lastMoved - 1) % this.state.list.length })
+        return
 
       case 40:
-      this.setState({ lastMoved: (this.state.lastMoved + 1) % this.state.list.length })
-      return
+        this.setState({ lastMoved: (this.state.lastMoved + 1) % this.state.list.length })
+        return
     }
   },
   render () {
     let schema = this.props.schema[0]
     let primaryField = getPrimaryField(schema)
     return (
       <ul className='list-group' onKeyDown={this.refocus}>
-        { this.state.list.length ?
-            this.state.list.map((item, i) => {
-              let editLink = FlowRouter.path('collection-item', {
-                projectId: this.props.routeParams.projectId,
-                pageId: this.props.routeParams.pageId,
-                collectionName: this.props.field,
-                index: i.toString()
-              })
+        {this.state.list.length
+            ? this.state.list.map((item, i) => {
```

Comment:
This mega ternary is kinda cool, but I think it'd be easier on the readers brain if we split it up like
```jsx
if (!this.state.list) {(
  <li className='list-group-item text-xs-center text-muted'>No entries</li>
)}
this.state.list.map(item, i) => { 
  // render item
}
```